File: SemiSupervisedPosTagger/SemiSupervisedPosTagger/inf_struct.cpp
//
// Created by Mohammad Sadegh Rasooli on 1/9/15.
// Copyright (c) 2015 Mohammad Sadegh Rasooli. All rights reserved.
//

#include "inf_struct.h"

inf_struct::inf_struct(averaged_perceptron perceptron) {
	averaged_weights=perceptron.get_averaged_weights();
	tag_size=perceptron.tag_size;
	feat_size=perceptron.feat_size;
}

inf_struct::~inf_struct() {
}

File: src/Structures/InfStruct.java
package Structures;

import Learning.AveragedPerceptron;

import java.io.Serializable;
import java.util.HashMap;

/**
 * Created by Mohammad Sadegh Rasooli.
 * ML-NLP Lab, Department of Computer Science, Columbia University
 * Date Created: 1/13/15
 * Time: 12:58 PM
 * To report any bugs or problems contact rasooli@cs.columbia.edu
 */

public class InfStruct implements Serializable {
    public HashMap<Integer, Float>[][] averagedWeights;
    public int tagSize;
    public int featSize;
    public int beamSize;
    public boolean useBeamSearch;

    public InfStruct(AveragedPerceptron perceptron,boolean useBeamSearch,int beamSize) {
        averagedWeights = perceptron.getAveragedWeights();
        tagSize = perceptron.tagSize();
        featSize = perceptron.featureSize();
        this.beamSize=beamSize;
        this.useBeamSearch=useBeamSearch;
    }
}
